### ISSUE TITLE
Issuer should be a top level constraint

### ISSUE THREAD

**Issue by tplooker** [decentralized-identity/presentation-exchange#211]
Currently with the examples given in the spec, the way in which credentials are matched with a list of trusted issuers is via a field level constraint. However, much like the arguments given in #151 this constraint would be better suited to be expressed more generally.

 ```
"constraints": {
          "fields": [
            {
              "path": [
                "$.issuer",
                "$.vc.issuer",
                "$.iss"
              ],
              "purpose": "We can only verify bank accounts if they are attested by a trusted bank, auditor or regulatory authority.",
              "filter": {
                "type": "string",
                "pattern": "did:example:123|did:example:456"
              }
            },
      ]
}
```

Changes to

```
"constraints": {
          "issuer": ["did:example:123", "did:example:456"],
          "fields": []
}
```

Why? Because the issuer is more than just a field level constraint in a credential it conveys who is the producer of the assertion or the issuer is and how to validate the assertion. How it is expressed in the request will effect other related utility such as the ability for a wallet to perform dynamic issuer discovery based on an inbound request from a relying party.

**Comment by Sakurann** [comment]
+1 to explicitly defining a new optional property to specify the `issuer` of the credential you are asking for.

one of the reasons during PE v1.0 discussions this property was avoided was for a fear of ending having a long list.
I would propose having a very clear language in a new text saying
- if NO restrictions to the issuer of the credential, leave out this property
- define syntax that you can specify 1/ concrete DID like `did:example:123`, and also 2/any DID from a particular DID method like `did:example`

I think we should also allow an HTTP URL to be listed as an issuer like "https://example.com/keys/foo.jwk"

**Comment by brentzundel** [comment]
I'm fine with making `issuer` a top-level constraint, and specifying that the value must be a URI

**Comment by David-Chadwick** [comment]
There is an alternative to specifying a long list of issuers, which could get unduly long e.g. if it were to be every bank that issues a VISA credit card; and that is to specify the trust federation that the issuer must belong to e.g. the.visa.network. To cater for this you would add "trust_federation" as an alternative to the "issuer" property

**Comment by JaceHensley** [comment]
I would be opposed to adding an `issuer` property to `constraints`. Because `issuer` only has meaning when talking about Verifiable Credentials, and Presentation Exchange doesn't only support VCs it doesn't care what it's moving around. As far as PE is concerned `$.issuer` is no different than `$.someOtherField`

**Comment by tlodderstedt** [comment]
@JaceHensley what other use cases beside VCs/VPs is PE used for in the wild?

**Comment by JaceHensley** [comment]
🤷 it doesn't really matter tho cause PE isn't for VCs/VPs specifically, it doesn't care what you throw at it but if you add a special `issuer` field all of a sudden it does

**Comment by tlodderstedt** [comment]
I don't think so since issuer is a general concept in identity.

**Comment by JaceHensley** [comment]
But it would mean something different in each format right ($.issuer vs $.issuer.id vs $.iss)? So if a new format came out that people wanted to use PE with then all implementations would have to be updated

**Comment by tlodderstedt** [comment]
I think it's completely the opposite. If there would be an abstract issuer field, this could be mapped to format specific meaning. So no need to change anything as long as an implementation does not support the new format/kind of assertion.

**Comment by Sakurann** [comment]
Yes, either it is a VC/VP, ID Token, mDL (honestly, what else?) there is a common concept of the "issuer" and the proposal is to have one way to specify the issuer by abstracting it.

**Comment by tlodderstedt** [comment]
I see the challenge that the PE processor needs to know where to look for the issuer value in a particular credential (e.g. iss in JWTs). This introduces format specific logic into the PE processing.

**Comment by csuwildcat** [comment]
> I see the challenge that the PE processor needs to know where to look for the issuer value in a particular credential (e.g. iss in JWTs). This introduces format specific logic into the PE processing.

Given we are going to add the per-Input Descriptor `format` prop, I am hoping we can retain the field-evaluated issuer check, given anyone who wants to isolate any issuer value by format can do so - is that OK with folks? (I believe the PE folks are onboard with this cc/ @JaceHensley @brentzundel @kimdhamilton)
